kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkConnector.java

```diff
@@ -86,4 +92,36 @@ public ConfigDef config() {
     return S3SinkConnectorConfig.getConfig();
   }
 
+  private boolean checkBucketExists(S3SinkConnectorConfig config) {
+    @SuppressWarnings("unchecked")
+    Class<? extends S3Storage> storageClass =
+            (Class<? extends S3Storage>)
+                    config.getClass(StorageCommonConfig.STORAGE_CLASS_CONFIG);
+
+    final S3Storage bucketCheckHelper = StorageFactory.createStorage(storageClass,
+            S3SinkConnectorConfig.class, config,
+            config.getString(StorageCommonConfig.STORE_URL_CONFIG));
+
+    return bucketCheckHelper.bucketExists();
+  }
+
+  @Override
+  public Config validate(final Map<String, String> connectorConfigs) {
+    ConfigDef configDef = this.config();
+    if (null == configDef) {
+      throw new ConnectException(String.format("%s.config() must return a ConfigDef that is not"
+              + " null.", this.getClass().getName()));
+    } else {
+
+      // Checking whether the bucket exists
+      if (!checkBucketExists(new S3SinkConnectorConfig(connectorConfigs))) {
+        throw new ConnectException("Bucket does not exist or invalid credentials "
+                + "provided");
+      }
+
+      List<ConfigValue> configValues = configDef.validate(connectorConfigs);
+      return new Config(configValues);
+    }
+  }
```

Comment:
not sure why we are overriding the `validate()` method here. Why can't the config class do that?

Reply:
This is because we need to check for the S3 bucket using the credentials provided by the user in `s3.credentials.provider.class` Config property. This is possible only after all the configs have been parsed. Please refer to this well written comment by Randall: [https://github.com/confluentinc/kafka-connect-storage-cloud/pull/302#discussion_r375069152](url)